File: ROS/day1/hw2/CpporPython_topic/hw2_py/hw2_py/P1.py
import rclpy
from rclpy.node import Node
from std_msgs.msg import String, Int32, Float32

class py_string(Node):
    def __init__(self):
        super().__init__('PY_PUB')
        self.string_publisher = self.create_publisher(String, 'pys', 10)
        self.int_publisher = self.create_publisher(Int32, 'pyi', 10)
        self.float_publisher = self.create_publisher(Float32, 'pyf',10)
        self.timer = self.create_timer(1, self.publish_string_msg)
        self.count = 0

    def publish_string_msg(self):
        msg_s = String()
        msg_s.data = 'From_PY'
        
        msg_i = Int32()
        msg_i.data = self.count

        msg_f = Float32()
        msg_f.data = 2.414

        self.string_publisher.publish(msg_s)
        self.int_publisher.publish(msg_i)
        self.float_publisher.publish(msg_f)
        self.get_logger().info('{0}'.format(msg_s.data))
        self.get_logger().info('{0}'.format(msg_i.data))
        self.get_logger().info('{0}'.format(msg_f.data))
        self.count += 1

def main(args=None):
    rclpy.init(args=args)
    node = py_string()
    try:
        rclpy.spin(node)
    except:
        node.get_logger().info('Keyboard Interrupt (Ctrl C)')
    finally:
        node.destroy_node()
        rclpy.shutdown()

if __name__ == '__main__':
    main()

File: ROS/day1/hw2/CpporPython_topic/hw2/src/P1.cpp
#include <chrono>
#include <functional>
#include <memory>
#include <string>
#include "rclcpp/rclcpp.hpp"

#include "std_msgs/msg/string.hpp"
#include "std_msgs/msg/int32.hpp"
#include "std_msgs/msg/float32.hpp"

using namespace std::chrono_literals;

class Publisher1 : public rclcpp::Node
{
  public:
    Publisher1()
    : Node("Cpp_PUB"), count_(0)
    {
      string_publisher_ = this->create_publisher<std_msgs::msg::String>("cpps", 10);
      int_publisher_ = this->create_publisher<std_msgs::msg::Int32>("cppi",10);
      float_publisher_ = this->create_publisher<std_msgs::msg::Float32>("cppf",10);
      timer_ = this->create_wall_timer(1s, std::bind(&Publisher1::timer_callback, this));
    }

  private:
    void timer_callback()
    {
      auto string1 = std_msgs::msg::String();
      string1.data = "From_Cpp";
      RCLCPP_INFO(this->get_logger(), "OUTPUT(S): '%s'", string1.data.c_str());
      string_publisher_->publish(string1);

      auto int1 = std_msgs::msg::Int32();
      int int1data = count_++;
      int1.data = int1data;
      RCLCPP_INFO(this->get_logger(), "OUTPUT(I): '%d'", int1.data);
      int_publisher_->publish(int1);

      auto float1 = std_msgs::msg::Float32();
      float1.data = 3.14f;
      RCLCPP_INFO(this->get_logger(), "OUTPUT(F): '%f'", float1.data);
      float_publisher_ ->publish(float1);
    }
    rclcpp::TimerBase::SharedPtr timer_;
    rclcpp::Publisher<std_msgs::msg::String>::SharedPtr string_publisher_;
    rclcpp::Publisher<std_msgs::msg::Int32>::SharedPtr int_publisher_;
    rclcpp::Publisher<std_msgs::msg::Float32>::SharedPtr float_publisher_;
    size_t count_;
};

int main(int argc, char * argv[])
{
  rclcpp::init(argc, argv);
  auto node = std::make_shared<Publisher1>();
  rclcpp::spin(node);
  rclcpp::shutdown();
  return 0;
}